.github/workflows/formatting.yml

```diff
@@ -0,0 +1,103 @@
+        with:
+          path: ~/.cargo/registry
+          key: cargo-registry
+
+      - name: Install dependencies
+        run: |
+          apt update
+          apt -y install \
+                libpcre3 \
+                libpcre3-dev \
+                build-essential \
+                autoconf \
+                automake \
+                git \
+                libtool \
+                libpcap-dev \
+                libnet1-dev \
+                libyaml-0-2 \
+                libyaml-dev \
+                libcap-ng-dev \
+                libcap-ng0 \
+                libmagic-dev \
+                libnetfilter-queue-dev \
+                libnetfilter-queue1 \
+                libnfnetlink-dev \
+                libnfnetlink0 \
+                libhiredis-dev \
+                libjansson-dev \
+                libpython2.7 \
+                make \
+                python \
+                rustc \
+                software-properties-common \
+                wget \
+                zlib1g \
+                zlib1g-dev
+      - name: Install packages for clang-format 9
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+          # no need to install full clang using https://apt.llvm.org/llvm.sh
+          # using clang 9
+          add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+          apt-get update
+          apt-get install -y clang-format-9
+      - name: Install cbindgen
+        run: cargo install --force --debug --version 0.14.1 cbindgen
+      - run: echo "::add-path::$HOME/.cargo/bin"
+      # If master has newer commits, github adds a merge commit from master into
+      # our branch.
+      # This screws up git clang-format as it'll format the merge as well and
+      # reports any formatting issues in those commits on master as well.
+      # We don't care about them here. So let's use the real branch HEAD commit.
+      - name: Checkout HEAD on branch, not merge commit!
+        uses: actions/checkout@v1
+        with:
+          ref: ${{ github.head_ref }}
+#          ref: ${{ github.event.pull_request.head.sha }}
+      - run: git clone https://github.com/OISF/libhtp -b 0.5.x
+      - run: ./autogen.sh
+      - run: ./configure --enable-unittests
+      - name: Check formatting
+        run: |
+          # Do not use make as it hides the script's exit code
+          # make diffstat-style-branch >> check_formatting_log.txt 2>&1
+          ./scripts/clang-format.sh check-branch --make --diffstat --show-commits  >> check_formatting_log.txt 2>&1
+          rc=$?
+          if [ $rc -eq 0 ]; then
+              cat check_formatting_log.txt
+              echo "Formatting is following code style guide"
+          elif [ $rc -eq 1 ]; then
+              # limit output as it might be a lot in the worst case
+              tail -n 100 check_formatting_log.txt
+              echo "::warning ::Formatting is not following code style guide!"
```

Comment:
It would be trivial to switch formatting problems to an error here.

Reply:
I'm not too keen on that right now.

Reply:
Can we make it an error but start out with having the job be allowed to fail? This way we'll get visual feedback, but it won't fail the build. On of our travis tests is set up this way. Not sure if github actions allows for it.

Reply:
Just played around a bit.

There's the ``continue-on-error`` one can set for a job which would allow us to set an "error" in the ``check-formatting`` job. However, the visual representation still sucks in github. It'll still set the job to green.

There's also a way to have a job fail and then still run follow-on jobs. However, that means configuring the follow-on jobs to still run if *any* of the previous jobs failed. No idea, how github decides on the order of jobs and dependencies...

Reply:
To be more precise, setting ``continue-on-error`` while not perfect is much better.

It'll show the "formatting-check" as failed in the pull-request's checks overview. See e.g. https://github.com/roligugus/suricata/pull/5

Compare that with the previous approach of a warning only and formatting-check always succeeded.

Reply:
Looks like that the ``continue-on-error`` setting is almost useless as it seems to only affect the yaml's jobs it is in. It could be interesting if ever a rustfmt/cargo fmt check was added into the same formatting-check.yaml though.

Anyhow, it seems one can pick which status checks that have to pass in your "Branch Protection Rules", see https://docs.github.com/en/github/administering-a-repository/enabling-required-status-checks

So as long as you don't require the formatting-check to pass, you can merge pull requests.

Hence, I'd recommend the formatting-check to error out in case of wrong formatting problems and not pick it as a required "status check" for merging purposes. That way, you'd see the red failure indication in the pull request, yet your merging is not blocked. I.e. it's up to you as maintainers to decide whether to request a formatting change on the pull request or merge something in anyways.

Reply:
I think that makes sense, we can always disable it again if we find it to be too intrusive.